spacy_curated_transformers/models/listeners.py

```diff
@@ -141,17 +134,39 @@ def build_scalar_weighting_listener_v1(
     grad_factor (float):
         Factor to multiply gradients with.
     """
-    transformer = ScalarWeightingListener(
+    listener = ScalarWeightingListener.new(
         upstream_name=upstream,
         weighting=weighting,
         pooling=pooling,
         width=width,
         grad_factor=grad_factor,
     )
-    return transformer
+    return listener
+
 
+# We need to store the listener state in the attributes dict of the listener model.
+# This custom class is used to ensure that this state is not serialized to disk.
+class _ListenerNonPersistentState:
+    batch_id: Optional[int] = None
+    outputs: Optional[TransformerModelOutput] = None
+    backprop: Optional[Callable[[List[List[Ragged]], Tuple[int]], Any]] = None
 
-class TransformerListener(Model):
+
+@serialize_attr.register(_ListenerNonPersistentState)
+def serialize_listener_non_persistent_state(
+    _, value: _ListenerNonPersistentState, name: str, model
+) -> bytes:
+    return bytes()
+
+
+@deserialize_attr.register(_ListenerNonPersistentState)
+def deserialize_listener_non_persistent_state(
+    _, value: bytes, name: str, model
+) -> _ListenerNonPersistentState:
+    return _ListenerNonPersistentState()
+
+
+class TransformerListener:
```

Comment:
I think the purpose of this class is easier to understand if we update the docstring to make it clear that it bundles some static methods to work with the listener state. Maybe the name should also be changed to something like `ListenerStateUtils`?

Reply:
Agreed; fixed.